packages/next/src/shared/lib/router/router.ts

```diff
@@ -1144,9 +1142,7 @@ export default class Router implements BaseRouter {
                 return true
               }
               handleHardNavigation({
-                url: addBasePath(
-                  addLocale(as, locale || this.locale, this.defaultLocale)
-                ),
+                url: addLocale(as, locale || this.locale, this.defaultLocale),
```

Comment:
This doesn't seem correct, as the `url` value here should be the full path including `basePath`

Reply:
@ijjk Before the change in line `1106` the condition in line `1111` never passed because the manifest doesn't include the `basePath` but `asNoSlash` did. After removing the `basePath` from `asNoSlash` this condition does passes but because `as` already has `basePath` it was being added twice.

My change here is to just to remove the addition of the `basePath` as maybe `as` always has it, but if some test fails it can be changed to check for the existence of the `basePath` before adding it so that it doesn't get duplicated.